src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -492,9 +508,15 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       return DevicePermission.PermissionGrantedByBrowser;
     }
 
-    const proposedConstraints: MediaStreamConstraints | null = this.calculateMediaStreamConstraints(
+    let proposedConstraints: MediaStreamConstraints | null = this.calculateMediaStreamConstraints(
```

Comment:
Cant we modify DeviceSelection and add groupId there? Then we do not need to maintain another mapping and need to do this either.